zipkin-cassandra/src/main/scala/com/twitter/cassie/connection/ClusterClientProvider.scala

```diff
@@ -141,7 +141,11 @@ private[cassie] class ClusterClientProvider(
         // call login
         val loginMap: java.util.Map[String, String] = HashMap("username" -> username, "password" -> password)
         val authRequest = new AuthenticationRequest(loginMap)
-        client.login(authRequest)
+        val future = client.login(authRequest)
+        future onFailure { exception =>
+          throw exception
+        }
```

Comment:
so, i can see where you might want to throw an exception here (if the call to `super.prepareConnFactory(keyspacedFactory)` doesn't raise an exception when `login` or `set_keyspace` return a `Future.exception`. I would make this a bit more explicit, however, in that case:

```scala
  class CassandraThriftFramedCodec(protocolFactory: TProtocolFactory, config: ClientCodecConfig)
      extends ThriftClientFramedCodec(protocolFactory: TProtocolFactory, config: ClientCodecConfig, None, false) {
    override def prepareConnFactory(factory: ServiceFactory[ThriftClientRequest, Array[Byte]]) = {
      val keyspacedSetFactory: ServiceFactory[ThriftClientRequest, Array[Byte]] = factory flatMap { service =>
        val client = new ServiceToClient(service, new TBinaryProtocol.Factory())
        val login = client.login(new AuthenticationRequest(Map("username" -> username, "password" -> password)))
        val keyspace = client.set_keyspace(keyspace)
        for {
          _ <- login.onFailure(throw _)
          _ <- keyspace.onFailure(throw _)
        } yield service
      }
      // set up tracing
      super.prepareConnFactory(keyspacedSetFactory)
    }
  }
```

Since both `.login` and `.set_keyspace` are of type `Future[Unit]`, so we just bind them to `_` and then make sure that we are explicit in the `.onFailure` calls so that the shape of the code shows you what's going to happen, otherwise we will `yield service` which will return a `Future[Service[...]]`.
I wouldn't be surprised if others disagree (looking at @mbbx6spp) or have a better way :)